README.md

```diff
@@ -37,6 +37,236 @@ flutterfire --help
+
+#### `--project` flag
+- selects the Firebase project and negates the command prompt to choose which Firebase project you wish to use.
+
+
+### Multi build configuration setup
+
+#### Android
+
+[Android Firebase setup](https://firebase.google.com/docs/projects/multiprojects#support_multiple_environments_in_your_android_application) is fairly straight forward. It requires the application ID and the file path where you want to place your `google-services.json` file. Imagine you want to setup you Flutter project's android release build configuration with its own Firebase app. Here is how it would look:
+
+```bash
+flutterfire configure --platforms=android --android-package-name=your.application.id --android-out=android/app/src/release/google-services.json
+```
+
+As we are using the `google-services` gradle plugin to package the `google-services.json` file with your app, you can create as many build types & build flavors you require by following this documentation on [adding the JSON file](https://developers.google.com/android/guides/google-services-plugin#adding_the_json_file). It is simply a case of matching the correct application ID you've associated with a Firebase app in your Firebase console with the correct file path.
+
+ **important**: Use the '--android-package-name' flag otherwise the FlutterFire CLI will auto-detect from your `build.gradle` file.
+
+#### Apple (iOS & macOS)
+ 
+Apple has a couple of different options depending on your project setup. You can configure your Firebase app with Apple platform using either a target or build configuration.
+
+**important**: Use the `--ios-bundle-id` and/or `--macos-bundle-id` flag otherwise the FlutterFire CLI will auto-detect from your `project.pbxproj` file.
+
+##### build configuration
+
+Most developers will likely configure with build configuration. As default, a Flutter app comes with three different build configurations: `Debug`, `Profile` & `Release` (**important**: FlutterFire CLI does not yet support renaming the target from `Runner` for **build configuration** which was a feature [released in Flutter 3.13.0](https://docs.flutter.dev/release/release-notes/release-notes-3.13.0)). Unlike android, it does not matter where you write the `GoogleService-Info.plist` file. Here is how to setup your Flutter Apple platform with build configuration:
+
+for iOS:
+```bash
+flutterfire configure --platforms=ios --ios-bundle-id=your.bundle.id --ios-build-config=Release --ios-out=ios/release/GoogleService-Info.plist
+```
+
+for macOS:
+```bash
+flutterfire configure --platforms=macos --macos-bundle-id=your.bundle.id --macos-build-config=Release --macos-out=macos/release/GoogleService-Info.plist
+```
```

Comment:
I think it would be helpful to mention that you need to re-run this command with every build configuration, so you need also to run it with `Profile` and `Debug`. This was something what me confused at the beginning because I assumed that FlutterFire detects the build configs and adds it for all build configs automatically.